### PR TITLE
Port fset

### PR DESCRIPTION
**Pull request by nickdrozd**

### CODE REVIEW

**Comment by shaleh** [comment]
Look at etc/DEBUG for suggestions to configure and remember to add --enable-rust-debug

File: rust_src/src/data.rs [hunk 1]
```diff
@@ -789,4 +791,38 @@ pub fn logcount(value: EmacsInt) -> i32 {
     unsafe { rust_count_one_bits(value as usize) }
 }
 
+/// Set SYMBOL's function definition to DEFINITION, and return DEFINITION.
+#[lisp_fn]
+pub fn fset(mut symbol: LispSymbolRef, definition: LispObject) -> LispObject {
+    /* Perhaps not quite the right error signal, but seems good enough.  */
+    let sym_obj = symbol.as_lisp_obj();
```
Comment (shaleh) [inline rust_src/src/data.rs]: Does leaving `symbol` a `LispObject` help?

(side note. We do not use C style comments. Replace /* */ with //.)

**Comment by nickdrozd** [comment]
Turns out I had a boolean flipped around `valid_lisp_object_p`.

File: rust_src/src/data.rs [hunk 2]
```diff
@@ -789,4 +791,38 @@ pub fn logcount(value: EmacsInt) -> i32 {
     unsafe { rust_count_one_bits(value as usize) }
 }
 
+/// Set SYMBOL's function definition to DEFINITION, and return DEFINITION.
+#[lisp_fn]
+pub fn fset(mut symbol: LispSymbolRef, definition: LispObject) -> LispObject {
+    // Perhaps not quite the right error signal, but seems good enough.
```
Comment (shaleh) [inline rust_src/src/data.rs]: Can we bring this comment down into the if?

File: rust_src/src/data.rs [hunk 3]
```diff
@@ -789,4 +791,39 @@ pub fn logcount(value: EmacsInt) -> i32 {
...
+
+    if is_autoload(function) {
+        put(symbol, Qautoload, function.as_cons_or_error().cdr());
+    }
+
+    // Convert to eassert or remove after GC bug is found.  In the
+    // meantime, check unconditionally, at a slight perf hit.
+    unsafe {
+        if valid_lisp_object_p(definition) == 0 {
+            emacs_abort();
```
Comment (brotzeit) [inline rust_src/src/data.rs]: We use `panic!` in rust.

Comment (shaleh) [reply rust_src/src/data.rs]: We should block `emacs_abort` from being ported by bindgen.

**Comment by nickdrozd** [comment]
Anything left on this one?

**Comment by nickdrozd** [comment]
Anything left on this one?

**Pull request merged**